src/ring_theory/ideal/operations.lean

```diff
@@ -1316,7 +1316,30 @@ lemma quotient_map_comp_mk {J : ideal R} {I : ideal S} {f : R →+* S} (H : J 
   (quotient_map I f H).comp (quotient.mk J) = (quotient.mk I).comp f :=
 ring_hom.ext (λ x, by simp only [function.comp_app, ring_hom.coe_comp, ideal.quotient_map_mk])
 
-/-- `H` and `h` are kept as separate hypothesis since H is used in constructing the quotient map -/
+/-- If `f : R ≃+* S` is a ring isomorphism and `I : ideal R`, then `map f (map f.symm) = I`. -/
+lemma map_of_equiv (I : ideal R) (f : R ≃+* S) : (I.map f.to_ring_hom).map f.symm.to_ring_hom = I :=
```

Comment:
Hmm, these statements are unfortunately not as readable as I would wish them to be. I think we should look for a way to make them a bit nicer.

Reply:
I don't see how to improve them, but any suggestion is appreciated :)

Reply:
[This zulip thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/simp-normal.20form.20for.20bundled.20hom.20conversion/near/226552809) is relevant, which would allow us to change this to use coercions instead.

Reply:
Why can you use coercions two lemmas below this one, but not here? It's the same situation, right?
Also, you could (partly) work around the issue by making these lemmas about two ideals `I` and `J` as well, just as for the equivs below. The lemmas would then also be more general.

Reply:
Sorry, for some reasons I forgot to answer. If I use 
```
lemma map_of_equiv (I : ideal R) (f : R ≃+* S) : (I.map (f : R →+* S)).map (f.symm : R →+* S) = I :=
by simp [map_map]
```
it doesn't work. @eric-wieser Are you working on changing some `simp` lemma to make this working?

Maybe I am missing something, but it seems very unnatural to me to introduce another ideal `J` in these lemmas... (while I see the point in the following equiv).

Reply:
> Are you working on changing some simp lemmas to make this working?

No, not actively. Currently there is no simp lemma between `(f : R →+* S)` and `f.to_ring_hom`, and it's fairly random whether lemmas are expressed in terms of one or the other.

Reply:
I think `simp` with `ring_equiv.to_ring_hom_eq_coe` (in reverse?) will allow you to use that statement though

Reply:
Yes, that works. I am not sure which one is better, but now there are coercions everywhere.